glean-core/rlb/src/private/memory_distribution.rs

```diff
@@ -0,0 +1,147 @@
+    /// and an `ErrorType::InvalidValue` error is recorded.
+    fn accumulate(&self, sample: u64) {
+        let metric = Arc::clone(&self.0);
+        dispatcher::launch(move || crate::with_glean(|glean| metric.accumulate(glean, sample)));
+    }
+
+    /// Accumulates the provided signed samples in the metric.
+    ///
+    /// This is required so that the platform-specific code can provide us with
+    /// 64 bit signed integers if no `u64` comparable type is available. This
+    /// will take care of filtering and reporting errors for any provided negative
+    /// sample.
+    ///
+    /// Please note that this assumes that the provided samples are already in
+    /// the "unit" declared by the instance of the implementing metric type
+    /// (e.g. if the implementing class is a [MemoryDistribution] and the
+    /// instance this method was called on is using kilobyte, then `samples` are
+    /// assumed to be in that unit).
+    ///
+    /// # Arguments
+    ///
+    /// * `samples` - The vector holding the samples to be recorded by the metric.
+    ///
+    /// ## Notes
+    ///
+    /// Discards any negative value in `samples` and report an `ErrorType::InvalidValue`
+    /// for each of them.
+    /// Values bigger than 1 Terabyte (2<sup>40</sup> bytes) are truncated
+    /// and an `ErrorType::InvalidValue` error is recorded.
+    fn accumulate_samples_signed(&self, samples: Vec<i64>) {
+        let metric = Arc::clone(&self.0);
+        dispatcher::launch(move || {
+            crate::with_glean(|glean| metric.accumulate_samples_signed(glean, samples))
+        });
+    }
+
+    /// **Exported for test purposes.**
+    ///
+    /// Gets the currently stored histogram as a DistributionData of the serialized value.
+    ///
+    /// This doesn't clear the stored value.
+    ///
+    /// # Arguments
+    ///
+    /// * `ping_name` - represents the optional name of the ping to retrieve the
+    ///   metric for. Defaults to the first value in `send_in_pings`.
+    fn test_get_value<'a, S: Into<Option<&'a str>>>(
+        &self,
+        ping_name: S,
+    ) -> Option<DistributionData> {
+        dispatcher::block_on_queue();
+
+        let queried_ping_name = ping_name
+            .into()
+            .unwrap_or_else(|| &self.0.meta().send_in_pings[0]);
+
+        crate::with_glean(|glean| self.0.test_get_value(glean, queried_ping_name))
+    }
+
+    /// **Exported for test purposes.**
+    ///
+    /// Gets the currently-stored histogram as a JSON String of the serialized value.
+    ///
+    /// This doesn't clear the stored value.
+    ///
+    /// # Arguments
+    ///
+    /// * `ping_name` - represents the optional name of the ping to retrieve the
+    ///   metric for. Defaults to the first value in `send_in_pings`.
+    fn test_get_value_as_json_string<'a, S: Into<Option<&'a str>>>(
+        &self,
+        ping_name: S,
+    ) -> Option<String> {
+        self.test_get_value(ping_name)
```

Comment:
Let's call `dispatcher::block_on_queue();` here as well!

Reply:
Even though `test_get_value` calls it for me?

Reply:
Oh, right. No need to do it here!